### ISSUE TITLE
Allow authenticated users on Academic Commons to not publish content

### ISSUE THREAD

**Issue by qbaron** [gwu-libraries/lai-drupal#28]
When an authenticated user creates content on the AC website, they automatically publish after clicking save. Ideally, they would have the ability to choose whether they are ready to publish or merely want to draft the content

**Comment by retrospec** [comment]
Drupal's core Workflow and Content Moderation modules can handle this issue. 

Will need to do the following:

- Enable core Workflow and Content Moderation modules
- Configure the generic Editorial workflow (Administration -> Configuration -> Workflow) to apply to all content types (or specific ones of choice; recommending all) via "THIS WORKFLOW APPLIES TO:" section

The basic workflow is already set to have states of draft, published and archived. This should be enough. However, the workflow can be extended or new instance created (if necessary).

**Comment by rdelaloy** [comment]
Is this issue related to content editors not being able to view/edit prescheduled content?

@retrospec can you give me an idea of how much time this fix would involve and if it might also fix the above issue?

**Comment by retrospec** [comment]
@rdelaloy Correct. The solution I've noted would fix the above issue (and related duplicates). The lift needed to implement this feature is minimal (under 15mins).

**Comment by maxstarkenburg** [comment]
Tried this out on my dev instance. It doesn't seem to resolve the issue of non-admins not being able to see pre-scheduled content (oh, never mind, I see there's another ticket (#5) and contrib module suggestion for that). Otherwise pretty neat.

Some follow-up questions:
* Going from "published" to "draft" doesn't unpublish the page, right? I'm still seeing the published version, which is fine, but I believe different from what happens on GW's Drupal, so just wanna be sure of things.
* I think the default moderation states are confusing when it comes to content using Scheduler's "Publish on" field. Should editors set things to "Draft" and Scheduler will handle the draft-to-published transition? I'm also not sure about potential weirdness with saving as "Published" something that Scheduler is set to publish at a future date (is that just redundant? will things blow up?).
* Is "archiving" the same as "unpublishing"? That seems to be what it essentially does, but wanna ask in case I'm missing some semantic implication, or if there's some weirdness with Scheduler or something. If it is, I'd prefer we edit the various state and transition labels to "Unpublish" and "Unpublished" instead of "Archive" and "Archived", which seems possible in the interface.
* It seems enabling moderation forces the "published by default" setting for content types to no longer stick (though apparently [they're planning on removing that option when moderation is enabled](https://www.drupal.org/project/drupal/issues/2990816) anyway)? Would be unfortunate, because this is a pet-peeve of mine with GW's Drupal.

Also, some steps to add:
* **New requirement:** we'd have to change permissions too. When I tried adding content as user `test-ac` (a non-admin account), I got some red error messages about not being able to save state Draft to Draft or something. I changed permissions to give Authenticated Users the various "Editorial workflow" permissions, which solved the problem.
* **New requirement**: though this is really to solve #16, looks like we also need to change their permissions to disable "Delete" in the various places non-admins currently can.
